extensions/cornerstone/src/Viewport/OHIFCornerstoneViewport.tsx

```diff
@@ -280,6 +281,25 @@ const OHIFCornerstoneViewport = React.memo(props => {
       viewportOptions.viewportType = STACK;
     }
 
+    if (needsRerendering) {
```

Comment:
So this block says, if it is `needsRerendering` it means it need to invalidate its stack? I'm confused since that was not the intention for having `needsRerendering` I beleive. The intension originally was when displaySetService metadata is changed via DisplaySetService.EVENTS.DISPLAY_SET_SERIES_METADATA_INVALIDATED, (if metadata is wrong, and user corrects it in the app at runtime) the data should gets invalidated. Can't you just put this logic in the same `useEffect` above?

Reply:
Yes, this block means to invalidate the stack if "needsRendering" is true. 

I am also confused because unless my search function is not working, I am not seeing `needsRendering` used anywhere in the code (although I believe I've seen it a while back when I implemented this).

Either way, I don't see why this property would be only tied to the "metadata" when it changes. In my case, I set this to `true` in the app (custom mode) after a process that involves setting hp, filtering,..., then I set this to true to let the viewport knows it should re-render. Maybe it also can be set to true when the metadata changes then you can maybe get rid off the event subscription but not sure.

For "DISPLAY_SET_SERIES_METADATA_INVALIDATED" it is now fired by the PanelPetSUV in the tmtv extention. 
Note that the viewer didn't use to "invalidate" a stack before anyway as the CacheService didn't support it until this PR.

Right now, it seems to me the two logics while kind of similar, they need to co-exist

Reply:
Yeah that was necessary a while back and unnecessary now. Ok, how about we change the event to `DISPLAYSET_INVALIDATED` ?

Reply:
None of this will be necessary if a NEW display set is created in the filter event  below.